File: lista 15questoes/15questoes em golang/att02.go
package main

import "fmt"

type Carro struct {
    marca     string
    modelo    string
    ano       int
    velocidade int
}

func (c *Carro) mostrarInformacao() {
    fmt.Printf("Marca: %s, Modelo: %s, Ano: %d\n", c.marca, c.modelo, c.ano)
}

func (c *Carro) acelerar(incremento int) {
    c.velocidade += incremento
}

func (c *Carro) pararCarro() {
    c.velocidade = 0
}

func (c Carro) mostraVelocidade() {
    if c.velocidade > 0 {
        fmt.Printf("Velocidade atual: %d km/h\n", c.velocidade)
    } else {
        fmt.Println("Carro parado")
    }
}

func main() {
    carro1 := Carro{"Mitsubishi", "Lancer", 2014}
    carro1.acelerar(50)
    carro1.mostrarInformacao()
    carro1.mostraVelocidade()
}

File: lista 15questoes/15 questoes em java/att02.java
class Carro {
    String marca;
    String modelo;
    int ano;
    int velocidade;

    Carro(String marca, String modelo, int ano) {
        this.marca = marca;
        this.modelo = modelo;
        this.ano = ano;
        this.velocidade = 0;
    }

    void mostrarInformacao() {
        System.out.println("Marca: " + marca + ", Modelo: " + modelo + ", Ano: " + ano);
    }

    void acelerar(int incremento) {
        velocidade += incremento;
    }

    void pararCarro() {
        velocidade = 0;
    }

    void mostraVelocidade() {
        System.out.println("Velocidade atual: " + velocidade + " km/h");
    }
}

class Main {
    public static void main(String[] args) {
        Carro carro1 = new Carro("Mitsubishi", "Lancer", 2014);
        carro1.acelerar(50);
        carro1.mostrarInformacao();
        carro1.mostraVelocidade();
    }
}